2-Authorization-I/1-call-graph/README.md

```diff
@@ -221,168 +227,242 @@ In **Azure AD**, the scopes (permissions) set directly on the application regist
+                                setError(error);
+                            }
+                        }
+                    }
                 }
-            });
-        }
-    }, [account, instance]);
-
-    if (accounts.length > 0) {
-        return (
-            <>
-                <span>There are currently {accounts.length} users signed in!</span>
-                {apiData && (<span>Data retrieved from API: {JSON.stringify(apiData)}</span>)}
-            </>
-        );
-    } else if (inProgress === "login") {
-        return <span>Login is currently in progress!</span>
-    } else {
-        return <span>There are currently no users signed in!</span>
-    }
-}
-```
-
-If `acquireTokenSilent()` fails, the recommended pattern is to fallback to one of the interactive methods i.e. `acquireTokenPopup()` or `acquireTokenRedirect()`. In the sample, each of these options are illustrated:
+            }
+        };
+        getToken();
+    }, [account, inProgress, instance]);
 
-- [Profile.jsx](./SPA/src/pages/Profile.jsx) uses `acquireTokenSilent` and falls back to `acquireTokenPopup` if interaction is required
-- [Tenant.jsx](./SPA/src/pages/Tenant.jsx) uses `acquireTokenSilent` and falls back to `acquireTokenRedirect` if interaction is required
-- [Mails.jsx](./SPA/src/pages/Mails.jsx) uses `acquireTokenRedirect` only.
+    return [response, error];
+};
 
-> :information_source: When using `acquireTokenRedirect`, you may want to set `navigateToLoginRequestUrl` in [msalConfig](./SPA/src/authConfig.js) to **true** if you wish to return back to the page where acquireTokenRedirect was called.
+export default useTokenAcquisition;
+```
 
 ### Access Token validation
 
 Clients should treat access tokens as opaque strings, as the contents of the token are intended for the **resource only** (such as a web API or Microsoft Graph). For validation and debugging purposes, developers can decode **JWT**s (*JSON Web Tokens*) using a site like [jwt.ms](https://jwt.ms).
 
-### Calling the Microsoft Graph API
-
-[Microsoft Graph JavaScript SDK](https://github.com/microsoftgraph/msgraph-sdk-javascript) provides various utility methods to query the Graph API. While the SDK has a default authentication provider that can be used in basic scenarios, it can also be extended to use with a custom authentication provider such as MSAL. To do so, we will initialize the Graph SDK client with [clientOptions](https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/dev/docs/CreatingClientInstance.md) method, which contains an `authProvider` object of class **MyAuthenticationProvider** that handles the token acquisition process for the client.
+### Checking for client capabilities
 
-```javascript
-export const getGraphClient = () => {
-    let clientOptions = {
-        authProvider: new MyAuthenticationProvider(),
-    };
+The client capabilities claim (xms_cc) indicate whether a client application can satisfy the claims challenge generated The following events:
```

Comment:
use the title and descriptions to as much extent as possible as laid out in [https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/blob/master/2-WebApp-graph-user/2-1-Call-MSGraph/README.md](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/blob/master/2-WebApp-graph-user/2-1-Call-MSGraph/README.md)